homeassistant/components/toon/config_flow.py

```diff
@@ -0,0 +1,159 @@
+"""Config flow to configure the Toon component."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
+
+from .const import (
+    CONF_CLIENT_ID, CONF_CLIENT_SECRET, CONF_DISPLAY, CONF_TENANT,
+    DATA_TOON_CONFIG, DEFAULT_TENANT, DOMAIN)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def configured_displays(hass):
+    """Returns a set of configured Toon displays."""
+    return set(
+        entry.data[CONF_DISPLAY]
+        for entry in hass.config_entries.async_entries(DOMAIN)
+    )
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class ToonFlowHandler(config_entries.ConfigFlow):
+    """Handle a Toon config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize the Toon flow."""
+        self.displays = None
+        self.username = None
+        self.password = None
+        self.tenant = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+
+        app = self.hass.data.get(DATA_TOON_CONFIG, {})
+
+        if not app:
+            return self.async_abort(reason='no_app')
+
+        return await self.async_step_authenticate(user_input)
+
+    async def _show_authenticaticate_form(self, errors=None):
+        """Shows the authentication form to the user."""
+        data_schema = vol.Schema({
+                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Optional(CONF_TENANT, default=DEFAULT_TENANT): str,
```

Comment:
I see that this defaults to `eneco`. What is the benefit of making this an option the user needs to deal with? Can we query their available tenants from the API given a username/password? Can we hardcode it to `eneco`, or make it a dropdown so that people can pick a known tenant?

Reply:
Eneco users are not informed about their `tenant` code as far as I know and has the largest user base (since, well, it is an Eneco product, 350.000 devices in that tenant now).
They are currently extending their reach and I know of 2 tenants (`electrabel` and `viesgo` in Spain). The API, unfortunately, does not provide a list, nor does their documentation.

Hence, I think the safest route would be to simply expose it.

Reply:
Addressed in the current version. It now is limited to 'eneco', 'electrabel' & 'viesgo' and provided as a dropdown so that people can pick their tenant easily.